packages/backend/src/services/UserService.test.ts

```diff
@@ -274,12 +285,110 @@ describe('UserService', () => {
             );
         });
-        test('should create user if not exists', async () => {
+        test('should create user', async () => {
+            // Mock that no identity is found for that email
+            (
+                openIdIdentityModel.findIdentitiesByEmail as jest.Mock
+            ).mockImplementationOnce(() => []);
+
             await userService.loginWithOpenId(openIdUser, undefined, undefined);
+            expect(
+                openIdIdentityModel.updateIdentityByOpenId as jest.Mock,
+            ).toHaveBeenCalledTimes(0);
+            expect(
+                openIdIdentityModel.createIdentity as jest.Mock,
+            ).toHaveBeenCalledTimes(0);
             expect(userModel.createUser as jest.Mock).toHaveBeenCalledTimes(1);
             expect(userModel.createUser as jest.Mock).toBeCalledWith(
                 openIdUser,
             );
+            expect(userModel.activateUser as jest.Mock).toHaveBeenCalledTimes(
+                0,
+            );
+        });
+        test('should activate invited user', async () => {
+            // Mock that no identity is found for that email
+            (
+                openIdIdentityModel.findIdentitiesByEmail as jest.Mock
+            ).mockImplementationOnce(() => []);
+
+            await userService.loginWithOpenId(
+                openIdUser,
+                undefined,
+                'inviteCode',
+            );
+            expect(
+                openIdIdentityModel.updateIdentityByOpenId as jest.Mock,
+            ).toHaveBeenCalledTimes(0);
+            expect(
+                openIdIdentityModel.createIdentity as jest.Mock,
+            ).toHaveBeenCalledTimes(0);
+            expect(userModel.createUser as jest.Mock).toHaveBeenCalledTimes(0);
+            expect(userModel.activateUser as jest.Mock).toHaveBeenCalledTimes(
+                1,
+            );
+        });
+        test('should link openid with authenticated user', async () => {
+            await userService.loginWithOpenId(
+                openIdUser,
+                authenticatedUser,
+                undefined,
+            );
+            expect(
+                openIdIdentityModel.updateIdentityByOpenId as jest.Mock,
+            ).toHaveBeenCalledTimes(0);
+            expect(
+                openIdIdentityModel.createIdentity as jest.Mock,
+            ).toHaveBeenCalledTimes(1);
+            expect(
+                openIdIdentityModel.createIdentity as jest.Mock,
+            ).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    userId: authenticatedUser.userId,
+                }),
+            );
+            expect(userModel.createUser as jest.Mock).toHaveBeenCalledTimes(0);
+            expect(userModel.activateUser as jest.Mock).toHaveBeenCalledTimes(
+                0,
+            );
+        });
+        test('should link openid to an existing user that has another OIDC with the same email', async () => {
```

Comment:
didn't we say this behaviour should be behind a feature flag ?

Reply:
Yes, adding in the next PR.